packages/web/src/components/gcds-pagination/gcds-pagination.tsx

```diff
@@ -0,0 +1,336 @@
+  watchCurrentPage(newValue) {
+    this.currentStep = newValue;
+  }
+
+  /**
+   * List display - URL object to create query strings and fragment on links
+   */
+   @Prop() url: Object;
+
+  /**
+   * Function to fire when pageChange event is called
+   */
+  @Prop() pageChangeHandler: Function;
+
+  @State() currentStep: number;
+
+  /*
+   * Events
+   */
+
+  /**
+   * Update value based on user input.
+   */
+  @Event() gcdsPageChange: EventEmitter<void>;
+
+  private onPageChange = (e) => {
+    if (this.pageChangeHandler) {
+      this.pageChangeHandler(e);
+    }
+
+    this.gcdsPageChange.emit();
+  }
+
+  /**
+   * Function to constuct <li> and <a> tags for display="list" pagination
+   */
+  private configurePaginationStep(page: number, end?: "next" | "previous" | null, mobile?: boolean) {
+
+    var linkAttrs = {
+      onClick: (e) => this.onPageChange(e),
+      href: this.url ? constructHref(this.el, page, end) : "javascript:void(0)",
+      "aria-label": !end ?
+        I18N[this.lang].pageNumberOf.replace('{#}', page).replace('{label}', this.label)
+      :
+        end == "next" ?
+          `${I18N[this.lang].nextPage}: ${I18N[this.lang].pageNumberOf.replace('{#}', ++page).replace('{label}', this.label)}`
+        :
+          `${I18N[this.lang].previousPage}: ${I18N[this.lang].pageNumberOf.replace('{#}', --page).replace('{label}', this.label)}`,
+    }
+
+    if (page == this.currentPage && !end) {
+      linkAttrs['aria-current'] = "page";
+    }
+
+    if (end == "next") {
+      return(<li><a  {...linkAttrs} class={!mobile ? "gcds-pagination-end-button" : "gcds-pagination-end-button-mobile"}>{I18N[this.lang].next}<gcds-icon class="gcds-pagination-arrow-right" name="arrow-right"></gcds-icon></a></li>);
+    }
+    else if (end == "previous") {
+      return(<li><a  {...linkAttrs} class={!mobile ? "gcds-pagination-end-button" : "gcds-pagination-end-button-mobile"}><gcds-icon class="gcds-pagination-arrow-left" name="arrow-left"></gcds-icon>{I18N[this.lang].previous}</a></li>);
+    }
+    else {
+      return (<li class={constructClasses(page, this.currentPage, this.totalPages)}><a {...linkAttrs}>{page}</a></li>);
+    }
+
+  }
+
+  /**
+   * Function to render the right steps for display="list" pagination
+   */
+  private configureListPagination() {
+    this.listitems = [];
+    this.mobilePrevNext = [];
+
+    if (this.currentPage != 1) {
+      this.listitems.push(this.configurePaginationStep(this.currentPage, "previous"));
+      this.mobilePrevNext.push(this.configurePaginationStep(this.currentPage, "previous", true));
+    }
+
+    // Flags to see if ellipses already rendered
+    var previousEllipses = false;
+    var nextEllipses = false;
+
+    for (var i = 1; i <= this.totalPages; i++) {
+
+      // Current page
+      if (i == this.currentPage) {
+        this.listitems.push(this.configurePaginationStep(i));
+      }
+      // Total pages less than 10
+      else if (this.totalPages < 10) {
```

Comment:
I think we can remove all these switch cases and handle the breakpoints with CSS instead.